tests/test_immutableset.py

```diff
@@ -209,7 +209,16 @@ def test_union(self):
         s2 = immutableset(["a", "c"])
         s3 = immutableset(["b", "c"])
         self.assertEqual(s1, s2.union(s3))
+        self.assertEqual(s1, s2.union(s3, check_top_type_matches=str))
         self.assertEqual(s1, s2 | s3)
+        # check union with plain iterable
+        self.assertEqual(s1, s2.union(["b", "c"]))
+
+        with self.assertRaises(RuntimeError):
+            # right hand provokes type failure
+            self.assertEqual(s1, s2.union([1, 2, 3], check_top_type_matches=str))
```

Comment:
For both of these, do you need the assertion in order to trigger the error?
```suggestion
            s2.union([1, 2, 3], check_top_type_matches=str)
```

Reply:
Fixed in 7283c8cfe075d9d72e0eb061e80ffd62ae2bc22e